youtube_dlc/downloader/hls.py

```diff
@@ -59,15 +59,14 @@ def can_download(manifest, info_dict):
         return all(check_results)
 
     def real_download(self, filename, info_dict):
-        allow_unplayable_formats = self._downloader.params.get('allow_unplayable_formats', False)
         man_url = info_dict['url']
         self.to_screen('[%s] Downloading m3u8 manifest' % self.FD_NAME)
 
         urlh = self.ydl.urlopen(self._prepare_url(info_dict, man_url))
         man_url = urlh.geturl()
         s = urlh.read().decode('utf-8', 'ignore')
 
-        if not self.can_download(s, info_dict) and allow_unplayable_formats is False:
+        if not self.params.get('allow_unplayable_formats') and not self.can_download(s, info_dict):
             if info_dict.get('extra_param_to_segment_url') or info_dict.get('_decryption_key_url'):
```

Comment:
No, can_download checks for supported encryption schemes.

Reply:
`can_download` checks for encryption as well as `is_live`. In the future, it could also check for any other cases for being un-downloadable. The bypass should be implemented only for the encryption case. You can move the check inside `can_download if my suggestion does not cover for all cases

Reply:
Fixed now